### DISCUSSION TITLE
login button not working on localhost

### DISCUSSION THREAD

**Discussion by Paglrommo** [NillionNetwork/discussions#136] [Q&A]
i am using vps but login button not working
![Screenshot 2024-10-24 023231](https://github.com/user-attachments/assets/bd0ad83a-5598-4a06-8e4a-f7c9987555fb)

**Comment by crypblizz8** [comment]
Did you end up reolving this?

**Comment by pintertamas** [comment]
I ran into this same issue on multiple platforms, does anyone know what could be the solution? I am following the guide step by step, but there must be something I misconfigured, because no matter what I try, the Login process just hangs up.

**Comment by wklausing** [comment]
Hey, I got this problem as well. I get this error:  Firefox can’t establish a connection to the server at ws://127.0.0.1:54936/.   

I get the same error on Chrome and devnet is running in a separate terminal.  Is the problem the wrong port? My guess from the outputs is that the port should be 48102 not 54936. But where can I configure this? I am running everything on MacOS

Here the devnet output:
nillion-devnet
ℹ️ cluster id is 9e68173f-9c23-4acc-ba81-4f079b639964
ℹ️ using 256 bit prime
ℹ️ storing state in /var/folders/qj/0knygmrs11x_rvlxnrz196n40000gn/T/.tmpMjSXaB (103.43Gbs available)
🏃 starting nilchain node in: /var/folders/qj/0knygmrs11x_rvlxnrz196n40000gn/T/.tmpMjSXaB/nillion-chain
⛓  nilchain JSON RPC available at http://127.0.0.1:48102
⛓  nilchain REST API available at http://localhost:26650
⛓  nilchain gRPC available at http://localhost:26649
🏃 starting node 1
🏃 starting node 2
🏃 starting node 3
👛 funding nilchain keys
📝 network configuration written to /Users/prabu/.config/nillion/networks/devnet.yaml
🌄 environment file written to /Users/prabu/.config/nillion/nillion-devnet.env


And here the next.js app output:
> nextjs@0.1.0 dev
> next dev

  ▲ Next.js 14.2.14
  - Local:        http://localhost:3000

 ✓ Starting...
 ✓ Ready in 6.3s
 ○ Compiling / ...
 ⚠ Circular dependency between chunks with runtime (_app-pages-browser_node_modules_nillion_client-wasm_dist_worker_js, webpack)
This prevents using hashes of each other and should be avoided.

Circular dependency between chunks with runtime (0, webpack-runtime)
This prevents using hashes of each other and should be avoided.
 GET / 200 in 21000ms
 ⚠ Circular dependency between chunks with runtime (_app-pages-browser_node_modules_nillion_client-wasm_dist_worker_js, webpack)
This prevents using hashes of each other and should be avoided.

Circular dependency between chunks with runtime (0, webpack-runtime)
This prevents using hashes of each other and should be avoided.
 GET /favicon.ico 200 in 7947ms
 GET / 200 in 753ms
 GET /favicon.ico 200 in 8ms
 GET / 200 in 36ms

**Reply by wklausing** [reply]
I tried a bit more and it seems the problem is the server. If someone finds a solutions, please let use know! 

wscat -c ws://127.0.0.1:54936/
error: Unexpected server response: 400

**Comment by crypblizz8** [comment]
Hey there, we are updating the package today so will ping you when it is ready! Apologies for that.

Could you confirm which version of nillion you are using by using this command in your shell/terminal:

`nillion -V`

**Comment by crypblizz8** [comment]
This should now be solved with the latest 

`npx @nillion/create-nillion-app`

Let me know if get stuck.

**Reply by pintertamas** [reply]
While installing the updated package, the process stalls during project creation, immediately following this log:

> Installing project dependencies...
> (node:118) ExperimentalWarning: CommonJS module /usr/lib/node_modules/npm/node_modules/debug/src/node.js is loading ES Module /usr/lib/node_modules/npm/node_modules/supports-color/index.js using require().
> Support for loading ES Module in require() is an experimental feature and might change at any time
> (Use `node --trace-warnings ...` to show where the warning was created)

I stopped the process after ~15 minutes. The project seems to be created, but only partially, because the server startup fails due to this error message:
```
> nextjs-020@0.1.0 dev
> next dev

sh: 1: next: not found
```

Please let me know how I could provide additional details!

**Reply by crypblizz8** [reply]
Could you share the whole log from the start of `npx @nillion/create-nillion-app` ? Thanks

**Reply by ALuhning** [reply]
Is this resolved?  Following the quickstart and when I hit login, get an error - looks like it's not querying the right port.

```
ℹ️ cluster id is 222257f5-f3ce-4b80-bdbc-0a51f6050996
ℹ️ using 256 bit prime
ℹ️ storing state in /tmp/.tmpjr62cZ (934.38Gbs available)
🏃 starting nilchain node in: /tmp/.tmpjr62cZ/nillion-chain
⛓  nilchain JSON RPC available at http://127.0.0.1:48102
⛓  nilchain REST API available at http://localhost:26650
⛓  nilchain gRPC available at http://localhost:26649
🏃 starting node 1
🏃 starting node 2
🏃 starting node 3
👛 funding nilchain keys
📝 network configuration written to /home/vitalpointai/.config/nillion/networks/devnet.yaml
🌄 environment file written to /home/vitalpointai/.config/nillion/nillion-devnet.env
```
![image](https://github.com/user-attachments/assets/5b7a67ef-ef46-43b3-8794-8dcf722ac527)

**Reply by ALuhning** [reply]
update - issue only occurs when specifying a seed as per the docs: ```nillion-devnet --seed my-seed```  If no seed, it works fine.  When there is a seed, found that I needed to provide the devnet.yaml bootnodeUrl created when the seeded devnet spins up to client-ts create-client.ts.  [Submitted PR for consideration](https://github.com/NillionNetwork/client-ts/pull/57) to enable bootnodeUrl to be passed as part of the connection config in loginButton.tsx like so: ```const client = await createClient({
        bootnodeUrl: "http://localhost:43552",
        network: NETWORK,
        seed: "my-seed"
      });```

**Comment by professionalcat** [comment]
is it ok now?
